pkg/accounts/service/fetchAccount.ts

```diff
@@ -0,0 +1,25 @@
+import { Option, Result } from 'mini-fn';
+import { AccountRepository } from '../model/repository.ts';
+import { Account } from '../model/account.ts';
+
+export class FetchAccountService {
+  private accountRepository: AccountRepository;
+
+  constructor(accountRepository: AccountRepository) {
+    this.accountRepository = accountRepository;
+  }
+
+  async fetchAccount(name: string): Promise<Result.Result<Error, Account>> {
+    const res = await this.accountRepository.findByName(name);
+    if (Option.isNone(res)) {
+      return Result.err(new Error('AccountNotFoundError'));
+    }
+    const account = Option.unwrap(res);
+
+    try {
+      return Result.ok(account);
+    } catch (e) {
+      return Result.err(e);
+    }
```

Comment:
こうした方がいいと思います(このままでは無意味な`try-catch`になってしまう).
```suggestion
    try {
      const account = Option.unwrap(res);
      return Result.ok(account);
    } catch (e) {
      return Result.err(e);
    }
```